src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -95,6 +102,21 @@ public void Dispose()
                     s_blendedNodesPool.ForgetTrackedObject(blendedTokens);
                 }
             }
+
+            var lexedTokens = _lexedTokens;
+            if (lexedTokens != null)
+            {
+                _lexedTokens = null;
+
+                // Put lexedTokens back into the pool if it's correctly sized.
+                if (lexedTokens.Length == CachedTokenArraySize)
+                {
+                    // Clear all written indexes in lexedTokens before releasing back to the pool
+                    Array.Clear(lexedTokens, 0, _maxWrittenLexedTokenIndex + 1);
+
+                    s_lexedTokensPool.Free(lexedTokens);
+                }
```

Comment:
Should we `ForgetTrackedObject` in the alternative (not putting lexedTokens back into the pool)?

Reply:
If it's not equal to this size, then it didn't come from the pool (as the pool will now only ever contain arrays of this length).